### ISSUE TITLE
NVDA not working with SPSS 24 x64's java access bridge when 'report object shortcut keys' is enabled

### ISSUE THREAD

**Issue by Neurrone** [nvaccess/nvda#7998]
#### Steps to reproduce:

I remember managing to get NVDA working with SPSS 24, which uses the java access bridge. However, a java update broke access bridge functionality, and I can't recall what I did a month ago to make it work, despite reading through the numerous duplicates.

#### Expected behavior:
NVDA should be able to read SPSS 24's UI, which uses java access bridge. It also doesn't read the java control panel, though without generating errors.

#### Actual behavior:

Instead, it errors every time I move through SPSS's UI:

```
Error in func event_stateChange from JABHandler.internalFunctionQueue
Traceback (most recent call last):
  File "queueHandler.pyc", line 50, in flushQueue
  File "JABHandler.pyc", line 648, in event_stateChange
  File "NVDAObjects\__init__.pyc", line 72, in __call__
  File "NVDAObjects\JAB\__init__.pyc", line 217, in __init__
  File "NVDAObjects\window\__init__.pyc", line 162, in __init__
ValueError: invalid or not specified window handle
ERROR - eventHandler.executeEvent (00:10:01.201):
error executing event: gainFocus on <NVDAObjects.JAB.Table object at 0x06FA2BD0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 152, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "C:\Users\Dickson\AppData\Roaming\nvda\addons\wintenApps\globalPlugins\wintenObjs\__init__.py", line 203, in event_gainFocus
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\JAB\__init__.pyc", line 527, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 942, in event_gainFocus
  File "NVDAObjects\JAB\__init__.pyc", line 475, in reportFocus
  File "NVDAObjects\__init__.pyc", line 830, in reportFocus
  File "speech.pyc", line 379, in speakObject
  File "speech.pyc", line 273, in speakObjectProperties
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 115, in _getPropertyViaCache
  File "NVDAObjects\JAB\__init__.pyc", line 243, in _get_keyboardShortcut
IndexError: invalid index
```

I've installed both the x86 and x64 versions of the JRE, and followed these [instructions to install the java access bridge](https://docs.oracle.com/javase/accessbridge/2.0.2/setup.htm) for both JREs (specifically, the "Installing Java Access Bridge for Windows 64-Bit" heading). Since the JAB is built into the JRE now, the only thing I had to do is to copy the WindowsAccessBridge-32 or 64 dlls into the respective folders. I also enabled both access bridges using `jabswitch /enable`. Both the `system32` and `syswow64` folders don't have the legacy `WindowsAccessBridge.dll`. I've also manually verified that the files that were supposed to be in both JREs are all there.

SPSS also has its own bundled JRE, which I also enabled access bridge for after the above failed to no avail.

#### System configuration:
NVDA version:
master-14851,e51f5080

NVDA Installed or portable:
installed

Other information:

Windows version:
Windows 10 1709 10.0.16299.192 x64

Name and version of other software in use when reproducing the issue:
SPSS 24, java 8 update 161 x86 and x64

#### Other questions:

Does the issue still occur after restarting your PC?
Yes

Have you tried any other versions of NVDA?
No

**Comment by leonardder** [comment]
Could you go to the object presentation settings and disable reporting of shortcuts from there?

**Comment by Neurrone** [comment]
@leonardder thanks a lot, that did the trick. Will change issue title to reflect accordingly. 

As a side note, it would be good to have instructions on getting access bridge working on x64 systems into the user guide - only knew about needing both x86 and x64 versions from Twitter.

**Comment by leonardder** [comment]
I've seen this issue before with SPSS, and I'm really not sure what causes this. I have SPSS 24 x64 running without problems here. Note that the SPSS JRE is slightly outdated.

**Comment by Neurrone** [comment]
@leonardder it works for you without turning off shortcut key detection?

**Comment by leonardder** [comment]
Yes.

**Comment by Adriani90** [comment]
@Neurrone are you still having this issue in SPSS 29.0 or 30.0 with NVDA 2024.4?

**Comment by Neurrone** [comment]
I no longer use this application, so I'll close this.
